plugins/inputs/redis/redis.go

```diff
@@ -253,8 +253,14 @@ func gatherInfoOutput(
 
 		val := strings.TrimSpace(parts[1])
 
+		// Some percentage values have a "%" suffix that we need to get rid of before int/float conversion
+		num_val := val
+		if strings.HasSuffix(num_val, "%") {
+			num_val = num_val[:len(num_val)-1]
+		}
```

Comment:
You can use `num_val := strings.TrimSuffix(val, "%")` here for a one-liner.

Reply:
Updated, thanks!